src/OpenTelemetry/ProcessorPipelineWeight.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Contains definitions for common processor pipeline weights.
+/// </summary>
+internal enum ProcessorPipelineWeight
+{
+    /// <summary>
+    /// Pipeline start. Value: <see cref="int.MinValue"/>.
+    /// </summary>
+    PipelineStart = int.MinValue,
```

Comment:
This is misleading - if there are multiple processors with this value, only one would be the start.

Reply:
Ya, it is more like ranges. This is why I called it "weight" and not "order". My thinking was if a bunch of registrations happened something like this...

```csharp
builder.AddProcessor(new ProcessorA() { PipelineWeight = ProcessorPipelineWeight.PipelineEnrichment });
builder.AddProcessor(new ProcessorB() { PipelineWeight = ProcessorPipelineWeight.PipelineStart });
builder.AddProcessor(new ProcessorC() { PipelineWeight = ProcessorPipelineWeight.PipelineExporter });
builder.AddProcessor(new ProcessorD() { PipelineWeight = ProcessorPipelineWeight.PipelineEnrichment });
builder.AddProcessor(new ProcessorE() { PipelineWeight = ProcessorPipelineWeight.PipelineStart });
```

...would yield a pipeline weighted as `ProcessorB -> ProcessorE -> ProcessorA -> ProcessorD -> ProcessorC` which would probably work for most users.

You can get absolute control if you _want_ for example this...

```csharp
builder.AddProcessor(new ProcessorA() { PipelineWeight = ProcessorPipelineWeight.PipelineEnrichment + 1 });
builder.AddProcessor(new ProcessorB() { PipelineWeight = ProcessorPipelineWeight.PipelineStart + 1 });
builder.AddProcessor(new ProcessorC() { PipelineWeight = ProcessorPipelineWeight.PipelineExporter });
builder.AddProcessor(new ProcessorD() { PipelineWeight = ProcessorPipelineWeight.PipelineEnrichment });
builder.AddProcessor(new ProcessorE() { PipelineWeight = ProcessorPipelineWeight.PipelineStart });
```

...yields a pipeline weighted as `ProcessorE -> ProcessorB -> ProcessorD -> ProcessorA -> ProcessorC`.

Reply:
I guess I know where you're coming from, probably no need to spend much time on this discussion since I feel we're trying to solve a simple problem by introducing a complex series of issues. Happy to discuss with you offline when #5400 is done.